packages/admin-frontend/pages/admin/statistics/most-read.js

```diff
@@ -0,0 +1,180 @@
+// @flow
+
+/**
+ * Part of GDL gdl-frontend.
+ * Copyright (C) 2019 GDL
+ *
+ * See LICENSE
+ */
+
+import * as React from 'react';
+
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableRow,
+  TableHead,
+  TableFooter,
+  Typography,
+  CircularProgress,
+  Button,
+  TextField,
+  FormControl,
+  Grid
+} from '@material-ui/core';
+import { SaveAlt as SaveIcon } from '@material-ui/icons';
+import type { BookStatistics } from '../../../types';
+import Container from '../../../components/Container';
+import Layout from '../../../components/Layout';
+import { fetchMostReadBooks } from '../../../lib/fetch';
+import formatMostReadDataToObjects from '../../../lib/utils';
+
+type State = {
+  mostReadBooks: Array<BookStatistics>,
+  loading: boolean,
+  numberOfBooksWanted: number,
+  periodOfDays: number
+};
+class MostRead extends React.Component<*, State> {
+  state = {
+    mostReadBooks: [],
+    loading: true,
+    numberOfBooksWanted: 0,
+    periodOfDays: 0
+  };
+
+  async componentDidMount() {
+    this.fetchResults();
+  }
+
+  handleChange = (name: string) => (event: any) => {
+    this.setState({
+      [name]: event.target.value
+    });
+  };
+
+  fetchResults = async () => {
+    const { numberOfBooksWanted, periodOfDays } = this.state;
+    this.setState({ loading: true });
+    const bookStatisticsRes = await fetchMostReadBooks(
+      numberOfBooksWanted,
+      periodOfDays
+    );
+
+    if (bookStatisticsRes.isOk) {
```

Comment:
Du setter bare resultatet hvis responsen er vellykket. Hvis den feiler eller at det er 0 datapunkter som returneres så vil den loade evig og du når ikke meldingen `Could not find any statistics on most read books.`

Reply:
Skrive om til
```javascript
this.setState({
      loading: false,
      mostReadBooks: bookStatisticsRes.isOk
        ? formatMostReadDataToObjects(bookStatisticsRes).data
        : []
    });
```